### PR TITLE
Update User by userId

### PR DESCRIPTION
**Pull request by EXR-97**

### CODE REVIEW

File: src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+     * Generic exceptions are handled.
+     *
+     * @param exception
+     * @return
+     */
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity handleGenericException(Exception exception) {
+        CommonResponse commonResponse = CommonResponse.builder().success(false)
+                .errorMessage(exception.getLocalizedMessage()).build();
+        return new ResponseEntity<>(commonResponse, HttpStatus.BAD_REQUEST);
```
Comment (kssreejith1994) [inline src/main/java/com/elixr/poc/common/exception/GlobalExceptionHandler.java]: Create a scenario that throws internal server error and check the status code returned from server in postman. Please let me know the result here

File: src/main/java/com/elixr/poc/common/util/MessagesUtil.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    private static final String MESSAGE_PROPERTY_FILE_PREFIX = "messages";
+
+    private static ResourceBundle moduleMessagesResourceBundle;
+
+    static {
+        try {
+            // Initialize messagesResourceBundle in a static block because the module may not have messages.properties.
+            moduleMessagesResourceBundle = ResourceBundle.getBundle(MESSAGE_PROPERTY_FILE_PREFIX);
+        } catch (MissingResourceException e) {
+            moduleMessagesResourceBundle = null;
```
Comment (kssreejith1994) [inline src/main/java/com/elixr/poc/common/util/MessagesUtil.java]: Why is the same file present in Divya's PR too? You all are working on different features right? And here a catch without log or rethrown exception should never be used.

File: src/main/java/com/elixr/poc/exception/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -16,18 +17,42 @@
...
+    /**
+     * Generic exceptions are handled.
+     * @param exception
+     * @return
+     */
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity handleGenericException(Exception exception) {
+        CommonErrorResponse commonErrorResponse = CommonErrorResponse.builder().success(false)
+                .errorMessage(exception.getLocalizedMessage()).build();
+        return new ResponseEntity<>(commonErrorResponse, HttpStatus.BAD_REQUEST);
```
Comment (kssreejith1994) [inline src/main/java/com/elixr/poc/exception/GlobalExceptionHandler.java]: Check for internal server error mentioned earlier here as well

File: src/main/java/com/elixr/poc/rest/controller/UserModificationController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+     * Calling a method updateUserPartially of service class.
+     * Returning the response.
+     * And handling the Exception if the userId is not matching.
+     * @param userId
+     * @param userDetails
+     * @return
+     */
+    @PatchMapping("/user/{userId}")
+    public ResponseEntity modifyUser(@PathVariable("userId") UUID userId, @RequestBody @Valid UserRequest userDetails){
+        HttpStatus httpStatus = HttpStatus.OK;
```
Comment (kssreejith1994) [inline src/main/java/com/elixr/poc/rest/controller/UserModificationController.java]: Return the httpstatus directly. Do not store it in another reference variable

File: src/main/java/com/elixr/poc/service/UserService.java [hunk 5]
```diff
@@ -21,53 +22,76 @@ public class UserService {
...
+
+    /**
+     * Finding User by userId and returning the user.
+     * @param userId
+     * @return
+     * @throws IdNotFoundException
+     */
+    public User getUserByUserId(UUID userId) throws IdNotFoundException {
+        Optional<User> user = userRepository.findById(userId);
+        return user.orElseThrow(() -> new IdNotFoundException(MessagesUtil.getMessage(MessagesKeyEnum.ENTITY_ID_DOES_NOT_EXISTS.getKey())));
```
Comment (kssreejith1994) [inline src/main/java/com/elixr/poc/service/UserService.java]: Please explain how does this line work

File: src/main/java/com/elixr/poc/util/PropertiesUtil.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package com.elixr.poc.util;
+
+import lombok.extern.log4j.Log4j2;
+import org.apache.catalina.core.Constants;
+
+import java.util.Properties;
+
+@Log4j2
```
Comment (kssreejith1994) [inline src/main/java/com/elixr/poc/util/PropertiesUtil.java]: Why is this annotation? Why not slf4j? What is the difference?

File: src/main/java/com/elixr/poc/util/PropertiesUtil.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+public class PropertiesUtil {
+    private static final String PATH = "/messages.properties";
```
Comment (kssreejith1994) [inline src/main/java/com/elixr/poc/util/PropertiesUtil.java]: Check how properties files can be loaded using configuration class in spring boot.https://www.baeldung.com/configuration-properties-in-spring-boot

File: src/main/resources/messages.properties [hunk 8]
```diff
@@ -0,0 +1,4 @@
+entity.deleted.successfully = Delete Successfully.
+entity.id.does.not.exists = ID Does Not Exists.
+entity.mandatory.field.missing = is missing.
+entity.user.exists = User already exists.
```
Comment (kssreejith1994) [inline src/main/resources/messages.properties]: Add new line.

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]

File: src/main/resources/messages.properties (hunk 8 above)
Comment (EXR-97) [reply src/main/resources/messages.properties]: Updated

File: src/main/java/com/elixr/poc/util/PropertiesUtil.java (hunk 7 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/util/PropertiesUtil.java]: Updated

File: src/main/java/com/elixr/poc/util/PropertiesUtil.java (hunk 6 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/util/PropertiesUtil.java]: Solved

File: src/main/java/com/elixr/poc/rest/controller/UserModificationController.java (hunk 4 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/rest/controller/UserModificationController.java]: Updated

File: src/main/java/com/elixr/poc/common/util/MessagesUtil.java (hunk 2 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/common/util/MessagesUtil.java]: Solved

File: src/main/java/com/elixr/poc/service/UserService.java (hunk 5 above)
Comment (EXR-97) [reply src/main/java/com/elixr/poc/service/UserService.java]: Searching the user in db by its userId, if user is present with the particular userId then returning the user, if user is not found then throwing an exception IdNotFoundException

**Review by mh-sraveendran — APPROVED** [review]

**Pull request merged**